File: clients/go/examples/test_tls_client.go
package main

import (
	"fmt"
	"log"
	"os"
	"time"

	blazecache "github.com/blazecache/client"
)

func main() {
	serverAddr := "localhost:8443"
	if len(os.Args) > 1 {
		serverAddr = os.Args[1]
	}

	fmt.Printf("Connecting to TLS server at %s...\n", serverAddr)

	// Wait a bit for server to be ready
	time.Sleep(2 * time.Second)

	// Create TLS client (insecure mode for self-signed cert)
	c, err := blazecache.NewTLSInsecure(serverAddr)
	if err != nil {
		log.Fatalf("Failed to create TLS client: %v", err)
	}

	// Test PING (not available in client, so we'll test GET/PUT/DELETE)
	fmt.Println("Testing TLS client operations...")

	// Test PUT
	item := &blazecache.Item{
		Key:   "test-key",
		Value: []byte("test-value"),
	}
	if err := c.Set(item); err != nil {
		log.Fatalf("PUT failed: %v", err)
	}
	fmt.Println("✓ PUT successful")

	// Test GET
	result, err := c.Get("test-key")
	if err != nil {
		log.Fatalf("GET failed: %v", err)
	}
	if result == nil {
		log.Fatalf("GET returned nil")
	}
	if string(result.Value) != "test-value" {
		log.Fatalf("GET returned wrong value: expected 'test-value', got '%s'", string(result.Value))
	}
	fmt.Printf("✓ GET successful: %s\n", string(result.Value))

	// Test DELETE
	if err := c.Delete("test-key"); err != nil {
		log.Fatalf("DELETE failed: %v", err)
	}
	fmt.Println("✓ DELETE successful")

	// Verify DELETE worked
	result, err = c.Get("test-key")
	if err == nil && result != nil {
		log.Fatalf("GET after DELETE should have failed or returned nil")
	}
	fmt.Println("✓ GET after DELETE correctly returned not found")

	fmt.Println("\n✅ All TLS tests passed!")
}

File: clients/java/src/main/java/com/blazecache/TestTLSClient.java
package com.blazecache;

import java.io.IOException;
import java.util.Arrays;
import java.util.Optional;

public class TestTLSClient {
    public static void main(String[] args) {
        String serverAddr = "localhost:8443";
        if (args.length > 0) {
            serverAddr = args[0];
        }

        System.out.println("Connecting to TLS server at " + serverAddr + "...");
        
        // Wait a bit for server to be ready
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }

        try {
            // Create TLS client (insecure mode for self-signed cert)
            TLSCacheClient client = new TLSCacheClient(
                Arrays.asList(serverAddr),
                TLSCacheClient.SelectionStrategy.ROUND_ROBIN,
                false // disable certificate verification for self-signed cert
            );

            System.out.println("Testing TLS client operations...");

            // Test PUT
            String key = "test-key";
            byte[] value = "test-value".getBytes();
            client.set(key, value);
            System.out.println("✓ PUT successful");

            // Test GET
            Optional<byte[]> result = client.get(key);
            if (result.isEmpty()) {
                System.err.println("✗ GET failed: returned empty");
                System.exit(1);
            }
            if (!Arrays.equals(result.get(), value)) {
                System.err.println("✗ GET returned wrong value");
                System.exit(1);
            }
            System.out.println("✓ GET successful: " + new String(result.get()));

            // Test DELETE
            boolean deleted = client.delete(key);
            if (!deleted) {
                System.err.println("✗ DELETE failed");
                System.exit(1);
            }
            System.out.println("✓ DELETE successful");

            // Verify DELETE worked
            result = client.get(key);
            if (result.isPresent()) {
                System.err.println("✗ GET after DELETE should have returned empty");
                System.exit(1);
            }
            System.out.println("✓ GET after DELETE correctly returned not found");

            System.out.println("\n✅ All TLS tests passed!");

        } catch (IOException e) {
            System.err.println("✗ Test failed: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}

